modules/acm_customer/src/Plugin/rest/resource/CustomerDeleteResource.php

```diff
@@ -36,8 +36,8 @@ class CustomerDeleteResource extends ResourceBase {
    *   HTTP Response.
    */
   public function post(array $data = []) {
```

Comment:
@malachy-mcconnell cc @miromichalicka @sdelbosc @joshirohit100 

this is a very old issue related to Drupal 8.5 upgrade, we can't have default values here for post

I checked by doing live dev and after changing it to following it worked
```
  public function post(array $data) {
```

I suggest we also add full message in logs when there is exception
```
    $this->logger->debug('CustomerDeleteResource message received: @message', [
      '@message' => json_encode($data),
    ]);
```

Reply:
Thank you for doing the research on this, I will adjust the PR as you suggest except we can't really log the customer data on failure as it raises PPI (personal private information) concerns notwithstanding that a fail implies there is no email data, but if we get an any customer details in a log archive it sets off PPI alarm bells.